ledger-service/http-json/src/main/scala/com/digitalasset/http/Endpoints.scala

```diff
@@ -499,6 +502,53 @@ class Endpoints(
     )(req)
       .map(ps => partiesResponse(parties = ps._1.toList, unknownParties = ps._2.toList))
 
+  def allocateUser(req: HttpRequest)(implicit
+      lc: LoggingContextOf[InstanceUUID with RequestID]
+  ): ET[domain.SyncResponse[Boolean]] =
+    proxyWithCommand { (jwt, createUserRequest: domain.CreateUserRequest) =>
+      {
+        import scalaz.std.list._
+        import scalaz.std.option._
+        import scalaz.syntax.traverse._
+        import scalaz.syntax.std.either._
+        import com.daml.lf.data.Ref
+        val input =
+          for {
+            username <- Ref.UserId.fromString(createUserRequest.userId).disjunction
+            primaryParty <- createUserRequest.primaryParty.traverse(it =>
+              Ref.Party.fromString(it).disjunction
+            )
+            canActAs <-
+              createUserRequest.canActAs.traverse(it =>
+                Ref.Party
+                  .fromString(it.toString)
+                  .map(CanActAs(_): UserRight)
+                  .disjunction
+              )
+            canReadAs <-
+              createUserRequest.canReadAs.traverse(it =>
+                Ref.Party.fromString(it.toString).map(CanReadAs(_): UserRight).disjunction
+              )
```

Comment:
`toString` should not be used to unwrap newtypes; it is nearly as bad as `asInstanceOf`.

```suggestion
              domain.Party
                .unsubst(createUserRequest.canActAs)
                .traverse(it =>
                  Ref.Party
                    .fromString(it)
                    .map(CanActAs(_): UserRight)
                    .disjunction
                )
            canReadAs <-
              domain.Party
                .unsubst(createUserRequest.canReadAs)
                .traverse(it => Ref.Party.fromString(it).map(CanReadAs(_): UserRight).disjunction)
```

Reply:
Many thanks for the insight